### PR TITLE
Fix getStyle returns null after adding a source

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: #634 
Fixes: https://github.com/mapbox/mapbox-maps-android/issues/583

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Fix getStyle returns null after adding a source</changelog>`.

### Summary of changes
`onStyleLoaded` will only be triggered once for one style, so if users add a new source to style after style loaded, `onStyleLoaded` will never be invoked. And during the loading time for this new source, `getStyle` will return null because `style.isStyleLoaded` will be false.
This pr add a property `styleLoaded` to record wheter the style is loaded or not no matter how many sources or layers are added after style loaded.

### CODE REVIEW

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 1]
```diff
@@ -52,6 +52,7 @@ class MapboxMap internal constructor(
   internal var isStyleLoadInitiated = false
   private val styleObserver = StyleObserver(this, nativeMapWeakRef, nativeObserver, pixelRatio)
   internal var renderHandler: Handler? = null
+  internal var styleLoaded = false
```
Comment (alexander-kulikovskii) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:55]: should it be with private setter?
because in Java `internal` converted into `public`?

Comment (Chaoba) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:55]: We need set it in test cases.

**Comment by kiryldz** [comment]
@Chaoba please link an issue that gets fixed by this PR.
Also we should have a new test for it that was failing before but should pass now.

**Comment by pengdev** [comment]
This PR reverts @tobrun's PR in https://github.com/mapbox/mapbox-maps-android/pull/332, where we used native `isStyleLoaded` instead of the platform side cached value. Would be good to get a clear understanding of the issue we experienced and check with gl-native team if we want to modify the behaviour of `isStyleLoaded` on the native side.

**Comment by pengdev** [comment]
Could be related to https://github.com/mapbox/mapbox-maps-android/issues/614 as well

**Comment by pengdev** [comment]
@Chaoba could you build a snapshot using this branch?

**Comment by tobrun** [comment]
> check with gl-native team if we want to modify the behaviour of isStyleLoaded on the native side.

This was already confirmed in the annotation related ticket that it's expected to return false when something new is loading. This is conform to gl-js implemenation. For our own internal style logic, we need to roll our own setup as proposed here.

**Review by pengdev — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Pull request merged**